src/app/components/Transactions/LogEvent.tsx

```diff
@@ -0,0 +1,137 @@
+  const theme = useTheme()
+  const isMobile = useMediaQuery(theme.breakpoints.down('sm'))
+  const { t } = useTranslation()
+  switch (event.type) {
+    case RuntimeEventType.coregas_used:
+      return <span>Gas used: {event.body.amount}</span>
+    case RuntimeEventType.evmlog:
+      return (
+        <>
+          <StyledDescriptionList titleWidth={isMobile ? '100px' : '200px'}>
+            <dt>{t('common.type')}</dt>
+            <dd>
+              <b>{event.evm_log_name}</b>
+            </dd>
+            {event.evm_log_params?.length && (
+              <>
+                {/*<dt>Params</dt>*/}
+                <dd>
+                  <Table sx={{ border: '1px solid lightgray' }}>
+                    <TableHead>
+                      <TableRow>
+                        <TableCell>{t('common.name')}</TableCell>
+                        <TableCell>{t('common.type')}</TableCell>
+                        <TableCell>{t('common.data')}</TableCell>
+                        <TableCell />
+                      </TableRow>
+                    </TableHead>
+                    <TableBody>
+                      {event.evm_log_params.map((param, index) => (
+                        <TableRow key={`param-${index}`}>
+                          <TableCell>{param.name}</TableCell>
+                          <TableCell>{param.evm_type}</TableCell>
+                          <TableCell>
+                            <EvmEventParamData param={param} layer={layer} />{' '}
+                          </TableCell>
+                          <TableCell>
+                            <CopyToClipboard
+                              label=" "
+                              value={
+                                typeof param.value === 'string'
+                                  ? (param.value as string)
+                                  : JSON.stringify(param.value, null, '  ')
+                              }
+                            />
+                          </TableCell>
+                        </TableRow>
+                      ))}
+                    </TableBody>
+                  </Table>
+                </dd>
+              </>
+            )}
+          </StyledDescriptionList>
+        </>
+      )
+
+    case RuntimeEventType.accountsburn:
+    case RuntimeEventType.accountsmint:
+    case RuntimeEventType.accountstransfer:
+    case RuntimeEventType.consensus_accountsdeposit:
+    case RuntimeEventType.consensus_accountswithdraw:
+    default:
+      return <pre>{JSON.stringify(event, null, ' ')}</pre>
+  }
+}
+
+export const TransactionLogEvent: FC<{ layer: Layer; event: RuntimeEvent; first: boolean }> = ({
+  layer,
+  event,
+  first,
+}) => {
+  const { t } = useTranslation()
+  const theme = useTheme()
+  const isMobile = useMediaQuery(theme.breakpoints.down('sm'))
+  const decoded = true // TODO: how do I know if this has been successfully decoded?
```

Comment:
API is currently not returning unparsed events. Does your typescript usage ensure that we will know if API starts returning new types of events?

Reply:
>  Does your typescript usage ensure that we will know if API starts returning new types of events?

Now it does.